### PR TITLE
(FACT-2987) Align hypervisors.zone.id fact on Solaris with Facter 3

### PR DESCRIPTION
**Pull request by luchihoratiu**
This commit converts the `hypervisors.zone.id` fact value into integer on Solaris (from string) to align with Facter 3 output.

### CODE REVIEW

**Comment by luchihoratiu** [comment]
jenkins please test this on solaris11-SPARC

**Comment by luchihoratiu** [comment]
jenkins please test this on solaris11-64,solaris11-SPARC,solaris114-64

**Comment by luchihoratiu** [comment]
jenkins please test this on solaris11-64,solaris11-SPARC,solaris114-64

**Review by GabrielNagy — APPROVED** [review]

File: spec/framework/utils/utils_spec.rb [hunk 1]
```diff
@@ -23,4 +23,18 @@
...
+    end
+
+    it 'does not convert string' do
+      expect(Facter::Utils.try_to_int('string')).to be('string')
+    end
+
+    it 'does not convert partial int' do
+      expect(Facter::Utils.try_to_int('7string')).to be('7string')
+    end
+  end
```
Comment (gimmyxd) [inline spec/framework/utils/utils_spec.rb:39]: Maybe add a couple of more tests, eg: integer, boolean, float value
```ruby
Facter::Utils.try_to_int(true)
Facter::Utils.try_to_int(7)
Facter::Utils.try_to_int(7.10)
Facter::Utils.try_to_int("7.10")
```

**Review by gimmyxd — APPROVED** [review]

**Pull request merged**